context/app/static/js/pages/CellsAPIDemo/CellPercentagesForDatasets.jsx

```diff
@@ -16,22 +17,16 @@ function CellPercentagesForDatasets(props) {
   const [message, setMessage] = useState(null);
 
   async function handleSubmit() {
-    const urlParams = new URLSearchParams();
-    uuids.split(',').forEach((uuid) => {
-      urlParams.append('uuid', uuid);
-    });
-    urlParams.append('gene_name', geneName);
-    urlParams.append('min_gene_expression', minGeneExpression);
-
-    const firstResponse = await fetch(`/cells/cell-percentages-for-datasets.json?${urlParams}`, {
-      method: 'POST',
-    });
-    const responseJson = await firstResponse.json();
-    if ('message' in responseJson) {
-      setMessage(responseJson.message);
-    }
-    if ('results' in responseJson) {
-      setResults(responseJson.results);
+    try {
+      setResults(
```

Comment:
I think I prefer to have the await statement on its own line and to call the setter after storing the data in a var.